botocore/response.py

```diff
@@ -45,6 +46,9 @@ def __init__(self, raw_stream, content_length):
         self._content_length = content_length
         self._amount_read = 0
 
+    def __del__(self):
```

Comment:
This seems a bit scary, this will disable garbage collection for the object right? Is that what we're actually trying to do?

Reply:
Implementing `__del__` allows you to extend behavior at GC time, having a stubbed implementation doesn't disable GC. There's a very specific scenario around circular references where Python assumes the presence of a `__del__` indicates custom cleanup logic and thus does not clean up the circularly referenced objects, but I believe this was fixed in Python 3.5+.